### PR TITLE
Support break inside DML 1.4 foreach statements -- SIMICS-18719

### PR DESCRIPTION
**Pull request by lwaern-intel**

### CODE REVIEW

File: py/dml/codegen.py [hunk 1]
```diff
@@ -121,7 +121,21 @@ def __init__(self):
 
     def break_(self, site):
         self.used = True
-        return [mkUnrolledBreak(site, self.label)]
+        return [mkGotoBreak(site, self.label)]
+
+class SequenceLoopContext(LoopContext):
```
Comment (mandolaerik) [inline py/dml/codegen.py:126]: unify context classes to, say, GotoLoopContext?

Comment (lwaern-intel) [reply py/dml/codegen.py:126]: They are so similar, in fact, I oopsied when I wrote `__init__` of `SequenceLoopContext`. I agree, they should be unified, but I don't really like the name `GotoLoopContext`.

Comment (lwaern-intel) [reply py/dml/codegen.py]: Pushed up a commit unifying them with the name `AbstractLoopContext` -- it's placeholder, I don't like it much more either. It's supposed to be "abstract" in the sense "a loop not corresponding to a C loop", but the name doesn't really communicate that.

**Comment by syssimics** [comment]
TeamCity build completed (SUCCESS): https://teamcity01-ir.devtools.intel.com/viewLog.html?buildId=9744625

**Comment by syssimics** [comment]
TeamCity build completed (SUCCESS): https://teamcity01-ir.devtools.intel.com/viewLog.html?buildId=9744624

File: py/dml/codegen.py (hunk 1 above)
Comment (mandolaerik) [reply py/dml/codegen.py]: `Abstract*` is usually an abstract class.

loop context is a thing that offers break. this class does that by gotoing to a label that the owner is responsible to provide. so i vote for a name with goto in.

Comment (lwaern-intel) [reply py/dml/codegen.py]: Alright. `GotoLoopContext` it is.

**Review by mandolaerik — APPROVED** [review]

**Pull request merged**